src/Silk.Core/Commands/Tests/TagCommand.cs

```diff
@@ -0,0 +1,270 @@
+            {
+                await ctx.RespondAsync(couldCreateAlias.reason);
+            }
+            else
+            {
+                await ctx.RespondAsync($"Alias `{aliasName}` that points to tag `{orignalTag}` successfuly created.");
+            }
+        }
+        
+        [Command]
+        public async Task Create(CommandContext ctx, string tagName, [RemainingText] string content)
+        {
+            Tag? tag = await _tagService.GetTagAsync(tagName, ctx.Guild.Id);
+            if (tag is not null)
+            {
+                await ctx.RespondAsync("Tag with that name already exists!");
+            }
+            if (_reservedWords.Any(tagName.StartsWith))
+            {
+                await ctx.RespondAsync("Tag name begins with reserved keyword!");
+                return;
+            }
+            if (string.IsNullOrEmpty(content))
+            {
+                await ctx.RespondAsync("Missing tag content!");
+                return;
+            }
+
+            var couldCreateTag = await _tagService.CreateTagAsync(tagName, content, ctx.Guild.Id, ctx.User.Id);
+            if (!couldCreateTag.success)
+            {
+                await ctx.RespondAsync(couldCreateTag.reason);
+            }
+            else
+            {
+                await ctx.RespondAsync($"Successfully created tag **{tagName}**.");
+            }
+        }
+
+        [Command]
+        public async Task Delete(CommandContext ctx, [RemainingText] string tagName)
+        {
+            if (string.IsNullOrEmpty(tagName))
+            {
+                await ctx.RespondAsync("Must specify a tag to delete!");
+                return;
+            }
+            
+            Tag? tag = await _tagService.GetTagAsync(tagName, ctx.Guild.Id);
+            if (tag is null)
+            {
+                await ctx.RespondAsync("Tag not found!");
+                return;
+            }
+            User? user = await _mediator.Send(new UserRequest.Get(ctx.Guild.Id, ctx.User.Id));
+
+            if (tag.OwnerId != ctx.User.Id && (!user?.Flags.Has(UserFlag.Staff) ?? true))
+            {
+                await ctx.RespondAsync("You either do not own this tag, or are not staff!");
+                return;
+            }
+
+            await _tagService.RemoveTagAsync(tagName, ctx.Guild.Id);
+            
+            string message = 
+                tag.OriginalTag is not null ?
+                "Alias successfully deleted." :
+                tag.Aliases?.Any() ?? false ?
+                    "Tag and all corresponded aliases successfully deleted." :
+                    "Tag successfully deleted.";
+            
+            await ctx.RespondAsync(message);
+
+        }
+
+        [Command]
+        public async Task Edit(CommandContext ctx, string tagName, [RemainingText] string content)
+        {
+            var couldEditTag = await _tagService.UpdateTagContentAsync(tagName, content, ctx.User.Id, ctx.User.Id);
```

Comment:
I made an oversight here. Second time I've done this too. Used the User Id instead of Guild Id, even though I'm looking directly at inlined variable names.